crates/biome_js_analyze/src/lint/nursery/no_useless_string_raw.rs

```diff
@@ -0,0 +1,113 @@
+use biome_analyze::{context::RuleContext, declare_lint_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::{AnyJsTemplateElement, JsTemplateExpression};
+use biome_rowan::{AstNode, AstNodeList};
+
+declare_lint_rule! {
+    /// Disallow unnecessary `String.raw` function in template string literals without any escape sequence.
+    ///
+    /// `String.raw` is useless when contains a raw string without any escape-like sequence.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// String.raw`a`;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// String.raw`a ${v}`;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// String.raw`\n ${a}`;
+    /// ```
+    ///
+    /// ```js
+    /// String.raw`\n`;
+    /// ```
+    pub NoUselessStringRaw {
+        version: "next",
+        name: "noUselessStringRaw",
+        language: "js",
+        recommended: false,
+    }
+}
+
+impl Rule for NoUselessStringRaw {
+    type Query = Ast<JsTemplateExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let tag = node.tag()?;
+        let tag = tag.as_js_static_member_expression()?;
+
+        let object = tag.object().ok()?;
+        let object_expr = object.as_js_identifier_expression()?;
+        let object_name = object_expr.name().ok()?.text();
+
+        let member = tag.member().ok()?;
+        let member_name = member.as_js_name()?.text();
+
+        if object_name != "String" || member_name != "raw" {
+            return None;
+        }
+
+        if can_remove_string_raw(node) {
+            Some(())
+        } else {
+            None
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "String.raw is useless when the raw string doesn't contain any escape sequence."
+                },
+            )
+            .note(markup! {
+                "Remove the String.raw call beacause it's useless here, String.raw can deal with string which contains escape sequence like \\n, \\t, \\r, \\\\, \\\", \\\'."
+            }),
+        )
+    }
+}
+
+fn contain_escape_sequences(input: &str) -> bool {
+    let mut chars = input.chars().peekable();
+    while let Some(ch) = chars.next() {
+        if ch == '\\' {
+            if let Some('n' | 't' | 'r' | '\\' | '"' | '\'' | '0') = chars.peek() {
+                return true;
+            }
+        }
+    }
+    false
```

Comment:
Iterating over `chars()` is expensive because this decodes UTF-8 characters.
Here we could directly operate over bytes.
Also, `peekable()` has some overhead that can be often avoided.
Here is my suggestion:

```suggestion
    let mut bytes = input.bytes();
    while let Some(c) = bytes.next() {
        if c == b'\\' {
            if matches!(bytes.next(), Some(b'n' | b't' | b'r' | b'\\' | b'"' | b'\'' | b'0')) {
                return true;
            }
        }
    }
    false
```

Actually I wonder even if we could just check for `\` because it will be directly interpreted in `String.raw` and regular strings. For instance in a regular string `\a` is a useless escape, however it will be results in the string `a`. While in a raw string this results in the string `\a`.
Thus, we could just check if the string contains `\`. What do you think?

```rs
    input.contains('\\')
```

Reply:
yes, i agree you, It is logically more reasonable to only check `\`